Add HRNet-w32-human-pose-estimation model (#2201)
openvinotoolkit/open_model_zoo: models/public/hrnet-w32-human-pose-estimation/hrnet-w32-human-pose-estimation.md

# hrnet-w32-human-pose-estimation

## Use Case and High-Level Description

The `HRNet-W32` model is one of the [HigherHRNet](https://arxiv.org/pdf/1908.10357).

> `HigherHRNet` is a novel bottom-up human pose

Reviewer: here you speak about HigherHRNet, while you name model without `higher` prefix in demo and models, possibly this prefix should be included into name?

Author: Add `higher` prefix

estimation method for learning scale-aware representations using high-resolution feature pyramids. The network uses HRNet as backbone, followed by one or more deconvolution modules to generate multi-resolution and high-resolution heatmaps. For every person in an image, the network detects a human pose: a body skeleton consisting of keypoints and connections between them. The pose may contain up to 17 keypoints: ears, eyes, nose, shoulders, elbows, wrists, hips, knees, and ankles.

> This is PyTorch\* implementation pretrained on COCO dataset.

Reviewer 2: ```suggestion
This is PyTorch\* implementation pre-trained on COCO dataset.
```

For details about implementation of model, check out the [HigherHRNet: Scale-Aware Representation Learning for Bottom-Up Human Pose Estimation](https://github.com/HRNet/HigherHRNet-Human-Pose-Estimation) repository.

## Specification

| Metric            | Value                  |
|-------------------|------------------------|
| Type              | Human pose estimation  |
| GFLOPs            | 92.8364                |
| MParams           | 28.6180                |
| Source framework  | PyTorch\*              |

## Accuracy

> | Metric | Original model | Converted model |
> | ------ | -------------- | --------------- |
> | Average Precision (AP)  | 65,02%          | 65,02%           |

Reviewer: ```suggestion
| Metric                            | Original model | Converted model |
| ------ -------------------| ----------------- | ------------------- |
| Average Precision (AP)  | 65,02%             | 65,02%                 |
```
P.S. I know that it  does not affect markdown representation, but it is more human-readable in editor mode

> | Average Precision (AP)     | 65,02%            | 65,02%          |

Reviewer 3: Was original model tested with original post-processing/decoder or with the one from OMZ?

> | Average Precision (AP)     | 65,02%            | 65,02%          |

Author: I add info about flip. Original model was tested on OMZ decoder (the conditions are the same for both models)

## Input

### Original Model

> Image, name - `image`,  shape - `1,3,512,512`, format is `B,C,H,W`, where:

Reviewer 2: ```suggestion
Image, name - `image`,  shape - `1, 3, 512, 512`, format is `B, C, H, W` where:
```

- `B` - batch size
- `C` - channel
- `H` - height
- `W` - width

> Channel order is `RGB`. Mean values - [123.675,116.28,103.53], scale values - [58.395,57.12,57.375].

Reviewer 2: ```suggestion
Channel order is `RGB`. Mean values - [123.675, 116.28, 103.53], scale values - [58.395, 57.12, 57.375].
```

### Converted Model

> Image, name - `image`,  shape - `1,3,512,512`, format is `B,C,H,W`, where:

Reviewer 2: ```suggestion
Image, name - `image`,  shape - `1, 3, 512, 512`, format is `B,C,H,W` where:
```

- `B` - batch size
- `C` - channel
- `H` - height
- `W` - width

Channel order is `BGR`.

## Output

The net outputs three blobs:

> - `heatmaps` of shape B, 17, 256, 256 containing location heatmaps for keypoints of all types.

Reviewer: ```suggestion
- `heatmaps` of shape `B, 17, 256, 256` containing location heatmaps for keypoints of all types.
```

Reviewer: what do you mean under all types?

Reviewer: B is batch size, right? If you have fixed batch size in input shape, I believe it does not change in output, so 1 should be instead of B

> - `nms_heatmaps` of shape B, 17, 256, 256 containing location heatmaps that are filtered out by non-maximum suppression algorithm.

Reviewer: ```suggestion
- `nms_heatmaps` of shape `1, 17, 256, 256` containing location heatmaps that are filtered out by non-maximum suppression algorithm.
```

> - `embeddings` of shape B, 17, 256, 256 containing associative embedding values, which are used for grouping individual keypoints into poses.

Reviewer: ```suggestion
- `embeddings` of shape `1, 17, 256, 256` containing associative embedding values, which are used for grouping individual keypoints into poses.
```

> - `embeddings` of shape `1, 17, 256, 256` containing associative embedding values, which are used for grouping individual keypoints into poses.

Author: Thank you for noticing, now I construct model as Intel hpe models with two outputs.
